pkg/cloud/amazon/s3_storage_test.go

```diff
@@ -396,6 +398,92 @@ func TestS3DisallowImplicitCredentials(t *testing.T) {
 	require.True(t, strings.Contains(err.Error(), "implicit"))
 }
 
+type awserror struct {
+	error
+	orig          error
+	code, message string
+}
+
+var _ awserr.Error = awserror{}
+
+func (a awserror) Code() string {
+	return a.code
+}
+
+func (a awserror) Message() string {
+	return a.message
+}
+
+func (a awserror) OrigErr() error {
+	return a.orig
+}
+
+func TestInterpretAWSCode(t *testing.T) {
+	{
+		// with code
+		input := awserror{
+			error: errors.New("hello"),
+			code:  s3.ErrCodeBucketAlreadyOwnedByYou,
+		}
+		got := interpretAWSError(input)
+		require.NotNil(t, got, "expected tryAWSCode to recognize an awserr.Error type")
+		require.False(t, errors.Is(got, cloud.ErrFileDoesNotExist), "should not include cloud.ErrFileDoesNotExist in the error chain")
+		require.True(t, strings.Contains(got.Error(), s3.ErrCodeBucketAlreadyOwnedByYou), "aws error code should be in the error chain")
+	}
+
+	{
+		// with keywords
+		input := awserror{
+			error: errors.New("‹AccessDenied: User: arn:aws:sts::12345:assumed-role/12345 is not authorized to perform: sts:AssumeRole on resource: arn:aws:iam::12345›"),
+		}
+		got := interpretAWSError(input)
+		require.NotNil(t, got, "expected interpretAWSError to recognize keywords")
+		require.True(t, strings.Contains(got.Error(), "AccessDenied"), "expected to see AccessDenied in error chain")
+		require.True(t, strings.Contains(got.Error(), "AssumeRole"), "expected to see AssumeRole in error chain")
```

Comment:
do you think it's worth adding something like this here and below?
```
require.Equal(t, "expected redacted error", errors.Redact(got))
```
it would be nice to see what redaction does exactly (probably we should do more of that in general..).

Reply:
Yep! Added.